.github/workflows/coverage.yml

```diff
@@ -1,68 +1,37 @@
+name: Coverage check
 
 on:
-  schedule:
-    # 04:00 daily
-    - cron: '0 4 * * *'
+  push:
+    branches:
+      - main 
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch: {}
 
 jobs:
-  check-coverage:
-    runs-on: ubuntu-latest
-    outputs:
-      msg: ${{ steps.make_msg.outputs.msg }}
+  coverage:
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os: [ubuntu]
     steps:
       - uses: actions/checkout@v4
-      - name: Install stable toolchain
-        uses: dtolnay/rust-toolchain@stable
-      - name: Install llvm
-        uses: KyleMayes/install-llvm-action@v1
+      - uses: mozilla-actions/sccache-action@v0.0.3
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          version: "16.0" # should match version used by rustc
-      - name: Download previous summary
-        uses: dawidd6/action-download-artifact@v2
-        with:
-          # Downloads the artifact from the most recent successful run
-          workflow: 'coverage.yml'
-          name: coverage-summary
-      - name: Rename file
-        run: mv coverage-summary.txt coverage-summary-prev.txt
-      - name: Generate coverage report and summary
-        env:
-          RUSTFLAGS: "-C instrument-coverage"
-        run: |
-          TGT=`cargo test --tests 2>&1 | grep Running | awk -F'[()]' '{print $2}'`
-          llvm-profdata merge -sparse default_*.profraw -o hugr.profdata
-          llvm-cov show --format=html --ignore-filename-regex='/.cargo/registry' --instr-profile=hugr.profdata --output-dir coverage --object ${TGT}
-          llvm-cov report --ignore-filename-regex='/.cargo/registry' --instr-profile=hugr.profdata --object ${TGT} | grep TOTAL | awk '{print $10}' | tr -dc '[:digit:].' > coverage-summary.txt
-      - name: Upload full report
-        uses: actions/upload-artifact@v3
-        with:
-          name: coverage
-          path: coverage/
-      - name: Upload summary
-        uses: actions/upload-artifact@v3
-        with:
-          name: coverage-summary
-          path: coverage-summary.txt
-      - name: Compare with previous summary and make message
-        id: make_msg
+          components: llvm-tools-preview
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Run tests with coverage instrumentation
         run: |
-          change="`cat coverage-summary-prev.txt`% --> `cat coverage-summary.txt`%"
-          if (( $(echo "`cat coverage-summary-prev.txt` < `cat coverage-summary.txt` + 0.04" | bc -l) ))
-          then
-            echo "msg=Coverage check for hugr shows no regression (${change}). ✅ ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> "$GITHUB_OUTPUT"
-          else
-            echo "msg=Coverage check for hugr shows regression (${change}). ❌ ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> "$GITHUB_OUTPUT"
-          fi
-  notify-slack:
```

Comment:
Is there a way we can still send a notification to slack with the headline number?

Reply:
I hoped the slack app would have this option, but I didn't find it.

I brought back the cron job (simplified a bit with cargo-llvm-cov). Instead of storing the full coverage result it now just links to the codecov page.
Unfortunately there is no web view for comparing two commits (that's only on the API), so I just link to the [landing page](https://app.codecov.io/gh/CQCL/hugr?search=&trend=7%20days) with a trend graph.